src/mam/mod_mam_meta.erl

```diff
@@ -232,9 +239,12 @@ add_rdbms_deps(user_cache, Type, #{cache_users := true, cache := CacheOpts}, Dep
                 mod_cache_users -> add_dep(mod_cache_users, Deps)
             end,
     add_dep(mod_mam_cache_user, #{Type => true, cache => CacheOpts}, Deps1);
-add_rdbms_deps(async_writer, Type, #{async_writer := AsyncOpts = #{enabled := true}}, Deps) ->
-    Deps1 = add_dep(rdbms_arch_module(Type), #{no_writer => true}, Deps),
-    add_dep(mod_mam_rdbms_arch_async, #{Type => AsyncOpts}, Deps1);
+add_rdbms_deps(async_writer, pm, #{async_writer := AsyncOpts = #{enabled := true}}, Deps) ->
+    Deps1 = add_dep(rdbms_arch_module(pm), #{no_writer => true}, Deps),
+    add_dep(mod_mam_rdbms_arch_async, AsyncOpts, Deps1);
+add_rdbms_deps(async_writer, muc, #{async_writer := AsyncOpts = #{enabled := true}}, Deps) ->
+    Deps1 = add_dep(rdbms_arch_module(muc), #{no_writer => true}, Deps),
+    add_dep(mod_mam_muc_rdbms_arch_async, AsyncOpts, Deps1);
```

Comment:
Why copy-paste? I worked hard to eliminate such duplications.
```suggestion
add_rdbms_deps(async_writer, Type, #{async_writer := AsyncOpts = #{enabled := true}}, Deps) ->
    Deps1 = add_dep(rdbms_arch_module(Type), #{no_writer => true}, Deps),
    add_dep(mod_mam_rdbms_arch_async, AsyncOpts, Deps1);
```

Reply:
Right, good sugestion 👌🏽